---
title: "Vocabulary for Chapter 6"
author: "Daniel Dean"
date: '2020-03-06'
slug: vocabulary-for-chapter-6
draft: no
categories:
  - Chapter 6
  - vocabulary
tags:
  - Chapter 6
  - vocabulary
subtitle: ''
summary: ''
authors: [daniel-dean]
lastmod: 
  featured: no
image:
  caption: ''
focal_point: ''
preview_only: no
projects: []
---

<script src="/rmarkdown-libs/kePrint/kePrint.js"></script>


<p>Chapter 6 covers Statistical Testing, including a review of null and alternative hypotheses (and associated distributions), types of error (I and II), as well as challenges and opportunities introduced by multiple testing.</p>
<table class="table table-striped table-hover table-condensed" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:right;">
</th>
<th style="text-align:left;">
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
Occam’s razor
</td>
<td style="text-align:left;width: 30em; ">
Heuristic stating that the simplest explanation for a phenomenon is often the best
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
rejection region
</td>
<td style="text-align:left;width: 30em; ">
Subset of possible outcomes for which probabilities under the null hypothesis fall under a low probability threshold, e.g. outcomes with a null-distribution probability &lt; 0.05; if an outcome falls within this region (e.g., p &lt; 0.05), it suggests that the null hypothesis is not true.
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
test statistic
</td>
<td style="text-align:left;width: 30em; ">
Metric for measuring how well a null hypothesis fits the data
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
null hypothesis
</td>
<td style="text-align:left;width: 30em; ">
Hypothesis describing some ‘uninteresting’ outcome (e.g., that no difference exists between certain groups of events/outcomes)
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
null distribution
</td>
<td style="text-align:left;width: 30em; ">
Distribution of possible outcomes, given that the null hypothesis is true
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
alternative hypothesis
</td>
<td style="text-align:left;width: 30em; ">
A hypothesis providing a different probability distribution than the null hypothesis; conceptually, holds that some difference from the null hypothesis exists (e.g. different means, frequencies, trends)
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
significance level/false positive rate/Type I error
</td>
<td style="text-align:left;width: 30em; ">
Probability of incorrectly rejecting the null hypothesis due to outcomes falling within the rejection region by chance; in terms of the null distribution, total probability that the outcome could fall within the rejection region given that the null hypothesis is true.
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
power
</td>
<td style="text-align:left;width: 30em; ">
True positive rate of a test (i.e., probability that an outcome falls in the rejection region of the null distribution, given that the alternative hypothesis is true)
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
false negative rate/Type II error
</td>
<td style="text-align:left;width: 30em; ">
Probability of incorrectly failing to reject the null hypothesis when an outcome from the alternative hypothesis distribution fails to fall within the rejection region of the null hypothesis.
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
specificity
</td>
<td style="text-align:left;width: 30em; ">
Complement of false positive rate (Type I error); probability of a test failing to reject the null hypothesis when it is true.
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
power/sensitivity/true negative rate
</td>
<td style="text-align:left;width: 30em; ">
Complement of false negative rate (Type II error); probability of correctly rejecting null hypothesis if the alternative hypothesis is true.
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
assumption of independence
</td>
<td style="text-align:left;width: 30em; ">
Treating every observation in a dataset as if it has no influence on the outcomes of other observations (or at least none unaccounted-for in the model).
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
p-value hacking
</td>
<td style="text-align:left;width: 30em; ">
Fallaciously ‘fishing’ for significant results by running tests until a small p-value is obtained by chance; this can be deliberate or inadvertently caused by a scattershot approach to testing.
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
hypothesis switching
</td>
<td style="text-align:left;width: 30em; ">
Fallacy of generating and/or changing hypotheses for a set of known results until a significant result is obtained by chance.
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
family-wide error rate (FWER)
</td>
<td style="text-align:left;width: 30em; ">
Probability of at least one false positive occurring in repeated tests. Assuming independent tests, this is the complement of the probability of only true positives occurring, and approaches 1.0 as the number of tests approaches infinity.
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
p-value histogram
</td>
<td style="text-align:left;width: 30em; ">
Visualization to get a quick sense of p-value distribution of possible test outcomes for a null hypothesis. The distribution is a mixture of cases where the null hypothesis is rejected (small p-values) or retained (larger p-values).
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
false discovery rate (FDR)
</td>
<td style="text-align:left;width: 30em; ">
The proportion of false positives among all cases where the null hypothesis is rejected across an entire distribution.
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
local false discovery rate (fdr)
</td>
<td style="text-align:left;width: 30em; ">
The probability of Type I Error at a given p-value when the distribution of the p-values is treated as a mixture model of the null distribution and alternative hypothesis distribution. This varies based on the p-value, rather than being a property of the entire distribution.
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
tail-area false disovery rate (Fdr)
</td>
<td style="text-align:left;width: 30em; ">
Integration-based extension of the local false discovery rate to obtain a false discovery rate for the entire distribution.
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
independent filtering
</td>
<td style="text-align:left;width: 30em; ">
Method to increase test power by filtering variables with criteria that are independent under the null hypothesis, but correlated under the alternative
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
independent hypothesis weighting
</td>
<td style="text-align:left;width: 30em; ">
A method of improving power of multiple testing by weighting hypotheses based on their power
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
Bonferroni adjustment
</td>
<td style="text-align:left;width: 30em; ">
Method used to compensate for inflated Type I (false positive) error in multiple testing by dividing the test significance level/hypothesis threshold (e.g., alpha = 0.05) by the number of tests performed
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
whole genome sequencing
</td>
<td style="text-align:left;width: 30em; ">
Method used to determine and record the DNA base values and order across all of an organism’s genes
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
marker gene
</td>
<td style="text-align:left;width: 30em; ">
A gene used to determine membership in a group of interest (e.g., a taxon, genotype within a population, or possessing a certain metabolic trait)
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
expression level
</td>
<td style="text-align:left;width: 30em; ">
The realtive abundance of transcriptions of a gene of interest present in, e.g., a cell or environment
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
reagent
</td>
<td style="text-align:left;width: 30em; ">
a compound used in creating a chemical reaction like an assay
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
hypothesis testing
</td>
<td style="text-align:left;width: 30em; ">
Evaluating whether outcomes are sufficently unlikely under the null hypothesis (holding that outcomes are determined fully by chance) that it can be rejected in favor of an alternative hypothesis
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
workflow
</td>
<td style="text-align:left;width: 30em; ">
A sequence of steps used in carrying out a larger operation or process
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
two-sided test
</td>
<td style="text-align:left;width: 30em; ">
A statistical test which rejects the null hypothesis if an observed test statistic is either too large or too small compared to that expected under the null hypothesis
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
one-sided test
</td>
<td style="text-align:left;width: 30em; ">
A statistical test which rejects the null hypothesis if an observed test statistic departs from the expected range in a single, predetermined direction (i.e. larger or smaller)
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
two-sample
</td>
<td style="text-align:left;width: 30em; ">
In the context of statistical testing, a situation whether the data belong to two known groups.
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
unpaired
</td>
<td style="text-align:left;width: 30em; ">
In the context of statistical tests, these are used when comparing groups with independent measurements (e.g. the observations for one group have no association with observations from the other group)
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
equal variances
</td>
<td style="text-align:left;width: 30em; ">
When groups being compared have (substantially) equivalent levels of variability.
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
dependence
</td>
<td style="text-align:left;width: 30em; ">
When the outcomes of two variables are associated with one another.
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
expected value
</td>
<td style="text-align:left;width: 30em; ">
For a random quantity, this is the value of the mean, i.e. “average value”.
</td>
</tr>
</tbody>
</table>
<div id="sources-consulted-or-cited" class="section level3">
<h3>Sources Consulted or Cited</h3>
<p><em>Some of the definitons above are based in part or whole on listed definitions in the following sources:</em></p>
<ul>
<li>Holmes and Huber, 2019. <em>Modern Statistics for Modern Biology.</em> Cambridge University Press,
Cambridge, United Kingdom.</li>
<li>Wikipedia: The Free Encyclopedia. <a href="http://en.wikipedia.org/wiki/Main_Page" class="uri">http://en.wikipedia.org/wiki/Main_Page</a></li>
<li>Bourgon, R., Gentleman, R. &amp; Huber, W. Independent filtering increases detection power for high-throughput experiments. Proceedings of the National Academy of Sciences 107, 9546–9551 (2010).</li>
<li>Ignatiadis, N., Klaus, B., Zaugg, J. et al. Data-driven hypothesis weighting increases detection power in genome-scale multiple testing. Nat Methods 13, 577–580 (2016). <a href="https://doi.org/10.1038/nmeth.3885" class="uri">https://doi.org/10.1038/nmeth.3885</a></li>
<li><a href="https://www.statisticssolutions.com/bonferroni-correction/" class="uri">https://www.statisticssolutions.com/bonferroni-correction/</a></li>
<li><a href="https://bioconductor.org/packages/release/bioc/vignettes/IHW/inst/doc/introduction_to_ihw.html" class="uri">https://bioconductor.org/packages/release/bioc/vignettes/IHW/inst/doc/introduction_to_ihw.html</a></li>
<li><a href="https://www.statisticshowto.datasciencecentral.com/familywise-error-rate/" class="uri">https://www.statisticshowto.datasciencecentral.com/familywise-error-rate/</a></li>
<li><a href="https://www.statisticshowto.datasciencecentral.com/probability-and-statistics/hypothesis-testing/" class="uri">https://www.statisticshowto.datasciencecentral.com/probability-and-statistics/hypothesis-testing/</a></li>
<li><p><a href="https://www.biostars.org/p/273537/" class="uri">https://www.biostars.org/p/273537/</a></p>
<h3 id="practice">Practice</h3>
<iframe src="https://quizlet.com/488864042/flashcards/embed?i=2oqpc3&amp;x=1jj1" height="500" width="100%" style="border:0">
</iframe></li>
</ul>
</div>
