---
title: "Chapter 8 Vocabulary List"
author: "Mikaela Elder"
date: '2020-04-06'
slug: chapter-8-vocabulary-list
categories:
  - vocabulary
  - chapter 8
tags:
  - vocabulary
  - chapter 8
subtitle: ''
summary: ''
authors: [mikaela-elder]
lastmod: '2020-04-06T16:16:47-06:00'
featured: no
image:
  caption: ''
  focal_point: ''
  preview_only: no
projects: []
---

<script src="/rmarkdown-libs/kePrint/kePrint.js"></script>


<p>Chapter 8 covers high-throughput count data, like data generated through RNA-seq. It introduces a number of tools that are useful for analyzing this type of data. The vocabulary terms for Chapter 8 are:</p>
<table class="table table-striped table-hover table-condensed" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:right;">
</th>
<th style="text-align:left;">
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
RNA-Seq
</td>
<td style="text-align:left;width: 30em; ">
sequencing of RNA molecules found in a population of cells or in a tissue
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
ChIP-Seq
</td>
<td style="text-align:left;width: 30em; ">
sequencing of DNA regions that are bound to particular DNA-binding proteins (selected by immunoprecipitation)
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
RIP-Seq
</td>
<td style="text-align:left;width: 30em; ">
sequencing of RNA molecules, or regions of them, bound to a particular RNA-binding protein
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
DNA-Seq
</td>
<td style="text-align:left;width: 30em; ">
sequencing of genomic DNA
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
HiC
</td>
<td style="text-align:left;width: 30em; ">
high-throughput chromatin conformation capture; a technique that aims to map the 3D spatial arrangement of DNA
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
cDNA
</td>
<td style="text-align:left;width: 30em; ">
complementary DNA made from RNA templates and reverse transcriptase; used in RNA-Seq
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
genetic screens
</td>
<td style="text-align:left;width: 30em; ">
a technique looking at the proliferation or survival of cells upon gene knockdown, knockout, or modification
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
read
</td>
<td style="text-align:left;width: 30em; ">
the sequence obtained from a fragment
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
sequencing library
</td>
<td style="text-align:left;width: 30em; ">
the collection of DNA molecules used as input for the sequencing machine
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
fragments
</td>
<td style="text-align:left;width: 30em; ">
molecules being sequenced during a sequencing analysis
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
count table
</td>
<td style="text-align:left;width: 30em; ">
a matrix with the tallies of the number of occurrences of subpopulations from a larger population/sample
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
dynamic range
</td>
<td style="text-align:left;width: 30em; ">
a ratio between the maximum and minimum values
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
heteroskedasticity
</td>
<td style="text-align:left;width: 30em; ">
a phenomenon where the variance and distribution shape of the data in different parts of the dynamic range are very different
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
normalization
</td>
<td style="text-align:left;width: 30em; ">
a technique that adjusts for the nature and magnitude of systematic sampling biases
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
rare events
</td>
<td style="text-align:left;width: 30em; ">
occurrences in the tail(s) of a distribution; observations that are extraordinarily high or low
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
dispersion
</td>
<td style="text-align:left;width: 30em; ">
a measure of the spread of the data; a common measure is the standard deviation or variance
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
gamma-Poisson
</td>
<td style="text-align:left;width: 30em; ">
negative binomial distribution with 2 parameters; ùõº and ùõΩ
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
systematic biases
</td>
<td style="text-align:left;width: 30em; ">
systematic distortions that affect the data generation and need to be accounted for in the analysis; one example would be variations in the total number of reads for each sample in a sequencing experiment
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
metadata
</td>
<td style="text-align:left;width: 30em; ">
a set of data that describes or gives information about other data
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
multifactorial design
</td>
<td style="text-align:left;width: 30em; ">
an experimental design with more than one independent variable
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
balanced
</td>
<td style="text-align:left;width: 30em; ">
in the context of study design, these are where there is an equal number of observations of all combinations of factors being tested
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
differential expression analysis
</td>
<td style="text-align:left;width: 30em; ">
a type of analysis that uses the normalized read count data to investigate quantitative changes in expression levels between different experimental groups
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
intercept
</td>
<td style="text-align:left;width: 30em; ">
a coefficient representing the base level of the measurement in the negative control
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
design factors
</td>
<td style="text-align:left;width: 30em; ">
binary indicator variables
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
interaction effect
</td>
<td style="text-align:left;width: 30em; ">
a parameter in a model that accounts for the effects of two experimental factors that combine in a more complicated fashion than a simple summation
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
design matrix
</td>
<td style="text-align:left;width: 30em; ">
a matrix encoding the design of an experiment where the columns correspond to experimental factors and the rows correspond to different experimental conditions
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
residuals
</td>
<td style="text-align:left;width: 30em; ">
a term in a model that reflects the experimental fluctuations (i.e.¬†random noise)
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
least sum-of-squares fitting
</td>
<td style="text-align:left;width: 30em; ">
a type of model fitting that minimizes the sum of the squared residuals
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
linear model
</td>
<td style="text-align:left;width: 30em; ">
a model that is a linear function of parameters, i.e.¬†takes the form: y_j = sum_k (x_jk * beta_k + e_j)
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
analysis of variance (ANOVA)
</td>
<td style="text-align:left;width: 30em; ">
an analysis that decomposes patterns in the data into systematic variability and noise
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
noise
</td>
<td style="text-align:left;width: 30em; ">
variability unaccounted for by model parameters
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
systematic variability
</td>
<td style="text-align:left;width: 30em; ">
variability accounted for by model parameters
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
breakdown point
</td>
<td style="text-align:left;width: 30em; ">
a measure of the robustness of an estimator; larger values indicate more robust estimators
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
robust
</td>
<td style="text-align:left;width: 30em; ">
a ‚Äústurdy‚Äù estimator that is not heavily influenced by outliers
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
least absolute deviations
</td>
<td style="text-align:left;width: 30em; ">
minimization of the sum of the absolute values of the residuals
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
least quantile of squares
</td>
<td style="text-align:left;width: 30em; ">
a type of regression where the difference between the model quantile and empirical quantile is minimized
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
least trimmed sum of squares
</td>
<td style="text-align:left;width: 30em; ">
a type of regression that minimized the sum of squared residuals, where the sum is over only a fraction of the smallest residuals
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
logistic regression
</td>
<td style="text-align:left;width: 30em; ">
a type of generalized linear regression for binary data where the outcome is transformed by the logistic function and bounded between 0 and 1
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
maximum likelihood
</td>
<td style="text-align:left;width: 30em; ">
a method for parameter estimation that finds the parameter value that maximizes the probablity of the observed data under the model
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
likelihood
</td>
<td style="text-align:left;width: 30em; ">
a function of a model parameter which is equal to the probability of the observed data under the model
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
maximum-likelihood estimates
</td>
<td style="text-align:left;width: 30em; ">
model parameters that are estimated by maximizing the probability of the observed data under the model
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
nuisance factor / blocking factor
</td>
<td style="text-align:left;width: 30em; ">
a factor that has some effect on the response but is of no interest to the experiment
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
batch effects
</td>
<td style="text-align:left;width: 30em; ">
hidden factors that affect the data but are not documented; e.g.¬†running samples at the same time have a degree of similarity from being run in the same batch
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
pseudocounts
</td>
<td style="text-align:left;width: 30em; ">
transformations that take the form y = log2(n + n_0) where n is the count and n_0 is a chosen positive constant
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
variance stabilizing transformation
</td>
<td style="text-align:left;width: 30em; ">
a transformation that has finite values and finite slope, even for counts close to zero
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
regularized logarithm (rlog) transformation
</td>
<td style="text-align:left;width: 30em; ">
a technique that transforms the original count data to a log2-like scale by fitting a ‚Äútrivial‚Äù model with a separate term for each sample and a prior distribution on the coefficients which is estimated from the data
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
Cook‚Äôs distance
</td>
<td style="text-align:left;width: 30em; ">
a measure of how much a single sample is influencing the coefficients in a model; large values indicate an outlier count
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
sampling without replacement
</td>
<td style="text-align:left;width: 30em; ">
a random sample in which no observation occurs more than one time in the sample
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
null hypothesis
</td>
<td style="text-align:left;width: 30em; ">
often, a hypothesis of ‚Äúno association‚Äù that is used as a counterpart to a more interesting alternative hypothesis in hypothesis testing.
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
variability
</td>
<td style="text-align:left;width: 30em; ">
in statistics, the amount by which a set of observations deviate from their mean
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
outlier
</td>
<td style="text-align:left;width: 30em; ">
a data point that does not follow the pattern of the rest of the data; often this data point will have a large residual
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
M-estimation
</td>
<td style="text-align:left;width: 30em; ">
a type of regression analysis that is more robust than OLS to outliers or data that does not follow a normal distribution; it minimizes the sum of the penalization function applied to the residuals
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
conservative
</td>
<td style="text-align:left;width: 30em; ">
an approach that prioritizes reducing false positives
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
splicing
</td>
<td style="text-align:left;width: 30em; ">
a process in eukaryotic organisms where mRNA is cut down from the full-length gene to just the exons before being translated
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
exons
</td>
<td style="text-align:left;width: 30em; ">
segments of a gene that actually get used during translation or encode for a protein
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
isoforms
</td>
<td style="text-align:left;width: 30em; ">
different forms of the same gene that result from splicing events that combine different exons in an mRNA script
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
upregulated
</td>
<td style="text-align:left;width: 30em; ">
a term used to describe the increased expression of a gene
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
gene knockdown
</td>
<td style="text-align:left;width: 30em; ">
a way of inactivating a gene by targeting its mRNA transcript for inactivation or degradation
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
gene knockout
</td>
<td style="text-align:left;width: 30em; ">
deletion of a gene from the genome
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
transcriptome
</td>
<td style="text-align:left;width: 30em; ">
the total of all of the mRNA expressed from genes in an organism
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
polymorphism
</td>
<td style="text-align:left;width: 30em; ">
genetic variation within a population
</td>
</tr>
</tbody>
</table>
<div id="sources-consulted-or-cited" class="section level3">
<h3>Sources consulted or cited</h3>
<p><em>Some of the definitions above are based in part or whole on listed definitions in the following sources.</em></p>
<ul>
<li>Holmes and Huber, 2019. <em>Modern Statistics for Modern Biology.</em> Cambridge University Press,
Cambridge, United Kingdom.</li>
<li>Lexico: <a href="https://www.lexico.com" class="uri">https://www.lexico.com</a></li>
<li>Statistics How To: <a href="https://www.statisticshowto.com" class="uri">https://www.statisticshowto.com</a></li>
<li>Lavrakas, 2008. Sampling without replacement. <em>Encyclopedia of Survey Research Methods.</em> <a href="https://dx.doi.org/10.4135/9781412963947.n516" class="uri">https://dx.doi.org/10.4135/9781412963947.n516</a></li>
</ul>
</div>
<div id="practice" class="section level3">
<h3>Practice</h3>
<iframe src="https://quizlet.com/501110916/flashcards/embed?i=2shf5e&amp;x=1jj1" height="500" width="100%" style="border:0">
</iframe>
</div>
