<!DOCTYPE html>
<html lang="en-us">

<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="Source Themes Academic 4.7.0">

  

  
  
  
  
  
    
    
    
  
  

  <meta name="author" content="Bailey Fosdick">

  
  
  
    
  
  <meta name="description" content="Chapter 8 covers high-throughput count data, like data generated through RNA-seq. It introduces a number of tools that are useful for analyzing this type of data. The vocabulary terms for Chapter 8 are:
        RNA-Seq  sequencing of RNA molecules found in a population of cells or in a tissue    ChIP-Seq  sequencing of DNA regions that are bound to particular DNA-binding proteins (selected by immunoprecipitation)    RIP-Seq  sequencing of RNA molecules, or regions of them, bound to a particular RNA-binding protein    DNA-Seq  sequencing of genomic DNA    HiC  high-throughput chromatin conformation capture; a technique that aims to map the 3D spatial arrangement of DNA    cDNA  complementary DNA made from RNA templates and reverse transcriptase; used in RNA-Seq    genetic screens  a technique looking at the proliferation or survival of cells upon gene knockdown, knockout, or modification    read  the sequence obtained from a fragment    sequencing library  the collection of DNA molecules used as input for the sequencing machine    fragments  molecules being sequenced during a sequencing analysis    count table  a matrix with the tallies of the number of occurrences of subpopulations from a larger population/sample    dynamic range  a ratio between the maximum and minimum values    heteroskedasticity  a phenomenon where the variance and distribution shape of the data in different parts of the dynamic range are very different    normalization  a technique that adjusts for the nature and magnitude of systematic sampling biases    rare events  occurrences in the tail(s) of a distribution; observations that are extraordinarily high or low    dispersion  a measure of the spread of the data; a common measure is the standard deviation or variance    gamma-Poisson  negative binomial distribution with 2 parameters; 𝛼 and 𝛽    systematic biases  systematic distortions that affect the data generation and need to be accounted for in the analysis; one example would be variations in the total number of reads for each sample in a sequencing experiment    metadata  a set of data that describes or gives information about other data    multifactorial design  an experimental design with more than one independent variable    balanced  in the context of study design, these are where there is an equal number of observations of all combinations of factors being tested    differential expression analysis  a type of analysis that uses the normalized read count data to investigate quantitative changes in expression levels between different experimental groups    intercept  a coefficient representing the base level of the measurement in the negative control    design factors  binary indicator variables    interaction effect  a parameter in a model that accounts for the effects of two experimental factors that combine in a more complicated fashion than a simple summation    design matrix  a matrix encoding the design of an experiment where the columns correspond to experimental factors and the rows correspond to different experimental conditions    residuals  a term in a model that reflects the experimental fluctuations (i.">

  
  <link rel="alternate" hreflang="en-us" href="/post/chapter-8-vocabulary-list/">

  


  
  
  
  <meta name="theme-color" content="#2962ff">
  

  
  
  
  <script src="/js/mathjax-config.js"></script>
  

  
  
  
  
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/all.min.css" integrity="sha256-+N4/V/SbAFiW1MPBCXnfnP9QSN3+Keu+NlB+0ev/YKQ=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.css" integrity="sha256-Vzbj7sDDS/woiFS3uNKo8eIuni59rjyNGtXfstRzStA=" crossorigin="anonymous">

    
    
    
      
    
    
      
      
        
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.10/styles/github.min.css" crossorigin="anonymous" title="hl-light">
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.10/styles/dracula.min.css" crossorigin="anonymous" title="hl-dark" disabled>
        
      
    

    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.5.1/leaflet.css" integrity="sha256-SHMGCYmST46SoyGgo4YR/9AlK1vf3ff84Aq9yK4hdqM=" crossorigin="anonymous">
    

    

    
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/lazysizes/5.1.2/lazysizes.min.js" integrity="sha256-Md1qLToewPeKjfAHU1zyPwOutccPAm5tahnaw7Osw0A=" crossorigin="anonymous" async></script>
      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml.js" integrity="" crossorigin="anonymous" async></script>
      
    
      

      
      

      
    

  

  
  
  
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Montserrat:400,700%7CRoboto:400,400italic,700%7CRoboto+Mono&display=swap">
  

  
  
  
  
  <link rel="stylesheet" href="/css/academic.css">

  





<script async src="https://www.googletagmanager.com/gtag/js?id=UA-36719749-3"></script>
<script>
  window.dataLayer = window.dataLayer || [];

  function gtag() {
      dataLayer.push(arguments);
  }

  function trackOutboundLink(url) {
    gtag('event', 'click', {
         'event_category': 'outbound',
         'event_label': url,
         'transport_type': 'beacon',
         'event_callback': function () {
           document.location = url;
         }
    });
    console.debug("Outbound link clicked: " + url);
  }

  function onClickCallback(event) {
    if ((event.target.tagName !== 'A') || (event.target.host === window.location.host)) {
      return;
    }
    trackOutboundLink(event.target);  
  }

  gtag('js', new Date());
  gtag('config', 'UA-36719749-3', {});

  
  document.addEventListener('click', onClickCallback, false);
</script>


  


  

  <link rel="manifest" href="/index.webmanifest">
  <link rel="icon" type="image/png" href="/images/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_32x32_fill_lanczos_center_2.png">
  <link rel="apple-touch-icon" type="image/png" href="/images/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_192x192_fill_lanczos_center_2.png">

  <link rel="canonical" href="/post/chapter-8-vocabulary-list/">

  
  
  
  
  
    
    
  
  
  <meta property="twitter:card" content="summary">
  
  <meta property="og:site_name" content="CSU MSMB Group Study">
  <meta property="og:url" content="/post/chapter-8-vocabulary-list/">
  <meta property="og:title" content="Chapter 8 Vocabulary List | CSU MSMB Group Study">
  <meta property="og:description" content="Chapter 8 covers high-throughput count data, like data generated through RNA-seq. It introduces a number of tools that are useful for analyzing this type of data. The vocabulary terms for Chapter 8 are:
        RNA-Seq  sequencing of RNA molecules found in a population of cells or in a tissue    ChIP-Seq  sequencing of DNA regions that are bound to particular DNA-binding proteins (selected by immunoprecipitation)    RIP-Seq  sequencing of RNA molecules, or regions of them, bound to a particular RNA-binding protein    DNA-Seq  sequencing of genomic DNA    HiC  high-throughput chromatin conformation capture; a technique that aims to map the 3D spatial arrangement of DNA    cDNA  complementary DNA made from RNA templates and reverse transcriptase; used in RNA-Seq    genetic screens  a technique looking at the proliferation or survival of cells upon gene knockdown, knockout, or modification    read  the sequence obtained from a fragment    sequencing library  the collection of DNA molecules used as input for the sequencing machine    fragments  molecules being sequenced during a sequencing analysis    count table  a matrix with the tallies of the number of occurrences of subpopulations from a larger population/sample    dynamic range  a ratio between the maximum and minimum values    heteroskedasticity  a phenomenon where the variance and distribution shape of the data in different parts of the dynamic range are very different    normalization  a technique that adjusts for the nature and magnitude of systematic sampling biases    rare events  occurrences in the tail(s) of a distribution; observations that are extraordinarily high or low    dispersion  a measure of the spread of the data; a common measure is the standard deviation or variance    gamma-Poisson  negative binomial distribution with 2 parameters; 𝛼 and 𝛽    systematic biases  systematic distortions that affect the data generation and need to be accounted for in the analysis; one example would be variations in the total number of reads for each sample in a sequencing experiment    metadata  a set of data that describes or gives information about other data    multifactorial design  an experimental design with more than one independent variable    balanced  in the context of study design, these are where there is an equal number of observations of all combinations of factors being tested    differential expression analysis  a type of analysis that uses the normalized read count data to investigate quantitative changes in expression levels between different experimental groups    intercept  a coefficient representing the base level of the measurement in the negative control    design factors  binary indicator variables    interaction effect  a parameter in a model that accounts for the effects of two experimental factors that combine in a more complicated fashion than a simple summation    design matrix  a matrix encoding the design of an experiment where the columns correspond to experimental factors and the rows correspond to different experimental conditions    residuals  a term in a model that reflects the experimental fluctuations (i."><meta property="og:image" content="img/map[gravatar:%!s(bool=false) shape:circle]">
  <meta property="twitter:image" content="img/map[gravatar:%!s(bool=false) shape:circle]"><meta property="og:locale" content="en-us">
  
    
      <meta property="article:published_time" content="2020-04-06T00:00:00&#43;00:00">
    
    <meta property="article:modified_time" content="2020-04-06T16:16:47-06:00">
  

  


    






  






<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "/post/chapter-8-vocabulary-list/"
  },
  "headline": "Chapter 8 Vocabulary List",
  
  "datePublished": "2020-04-06T00:00:00Z",
  "dateModified": "2020-04-06T16:16:47-06:00",
  
  "author": {
    "@type": "Person",
    "name": "Mikaela Elder"
  },
  
  "publisher": {
    "@type": "Organization",
    "name": "CSU MSMB Group Study",
    "logo": {
      "@type": "ImageObject",
      "url": "img//"
    }
  },
  "description": "Chapter 8 covers high-throughput count data, like data generated through RNA-seq. It introduces a number of tools that are useful for analyzing this type of data. The vocabulary terms for Chapter 8 are:\n        RNA-Seq  sequencing of RNA molecules found in a population of cells or in a tissue    ChIP-Seq  sequencing of DNA regions that are bound to particular DNA-binding proteins (selected by immunoprecipitation)    RIP-Seq  sequencing of RNA molecules, or regions of them, bound to a particular RNA-binding protein    DNA-Seq  sequencing of genomic DNA    HiC  high-throughput chromatin conformation capture; a technique that aims to map the 3D spatial arrangement of DNA    cDNA  complementary DNA made from RNA templates and reverse transcriptase; used in RNA-Seq    genetic screens  a technique looking at the proliferation or survival of cells upon gene knockdown, knockout, or modification    read  the sequence obtained from a fragment    sequencing library  the collection of DNA molecules used as input for the sequencing machine    fragments  molecules being sequenced during a sequencing analysis    count table  a matrix with the tallies of the number of occurrences of subpopulations from a larger population/sample    dynamic range  a ratio between the maximum and minimum values    heteroskedasticity  a phenomenon where the variance and distribution shape of the data in different parts of the dynamic range are very different    normalization  a technique that adjusts for the nature and magnitude of systematic sampling biases    rare events  occurrences in the tail(s) of a distribution; observations that are extraordinarily high or low    dispersion  a measure of the spread of the data; a common measure is the standard deviation or variance    gamma-Poisson  negative binomial distribution with 2 parameters; 𝛼 and 𝛽    systematic biases  systematic distortions that affect the data generation and need to be accounted for in the analysis; one example would be variations in the total number of reads for each sample in a sequencing experiment    metadata  a set of data that describes or gives information about other data    multifactorial design  an experimental design with more than one independent variable    balanced  in the context of study design, these are where there is an equal number of observations of all combinations of factors being tested    differential expression analysis  a type of analysis that uses the normalized read count data to investigate quantitative changes in expression levels between different experimental groups    intercept  a coefficient representing the base level of the measurement in the negative control    design factors  binary indicator variables    interaction effect  a parameter in a model that accounts for the effects of two experimental factors that combine in a more complicated fashion than a simple summation    design matrix  a matrix encoding the design of an experiment where the columns correspond to experimental factors and the rows correspond to different experimental conditions    residuals  a term in a model that reflects the experimental fluctuations (i."
}
</script>

  

  


  


  





  <title>Chapter 8 Vocabulary List | CSU MSMB Group Study</title>

</head>

<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" >

  <aside class="search-results" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search">
        
      </div>

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>


  







<nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id="navbar-main">
  <div class="container">

    
    <div class="d-none d-lg-inline-flex">
      <a class="navbar-brand" href="/">CSU MSMB Group Study</a>
    </div>
    

    
    <button type="button" class="navbar-toggler" data-toggle="collapse"
            data-target="#navbar-content" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
    <span><i class="fas fa-bars"></i></span>
    </button>
    

    
    <div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none">
      <a class="navbar-brand" href="/">CSU MSMB Group Study</a>
    </div>
    

    
    
    <div class="navbar-collapse main-menu-item collapse justify-content-start" id="navbar-content">

      
      <ul class="navbar-nav d-md-inline-flex">
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#people"><span>People</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#posts"><span>Posts</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#experience"><span>Schedule</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#contact"><span>Location</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/files/csu_msmb_syllabus.pdf"><span>Syllabus</span></a>
        </li>

        
        

      

        
      </ul>
    </div>

    <ul class="nav-icons navbar-nav flex-row ml-auto d-flex pl-md-2">
      
      <li class="nav-item">
        <a class="nav-link js-search" href="#"><i class="fas fa-search" aria-hidden="true"></i></a>
      </li>
      

      
      <li class="nav-item">
        <a class="nav-link js-dark-toggle" href="#"><i class="fas fa-moon" aria-hidden="true"></i></a>
      </li>
      

      

    </ul>

  </div>
</nav>


  <article class="article">

  












  

  
  
  
<div class="article-container pt-3">
  <h1>Chapter 8 Vocabulary List</h1>

  

  
    


<div class="article-metadata">

  
  
  
  
  <div>
    



  
  <span><a href="/authors/mikaela-elder/">Mikaela Elder</a></span>

  </div>
  
  

  
  <span class="article-date">
    
    
      
          Last updated on
      
    
    Apr 6, 2020
  </span>
  

  

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    6 min read
  </span>
  

  
  
  

  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fas fa-folder mr-1"></i><a href="/categories/vocabulary/">vocabulary</a>, <a href="/categories/chapter-8/">chapter 8</a></span>
  

</div>

    














  
</div>



  <div class="article-container">

    <div class="article-style">
      
<script src="/rmarkdown-libs/kePrint/kePrint.js"></script>


<p>Chapter 8 covers high-throughput count data, like data generated through RNA-seq. It introduces a number of tools that are useful for analyzing this type of data. The vocabulary terms for Chapter 8 are:</p>
<table class="table table-striped table-hover table-condensed" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:right;">
</th>
<th style="text-align:left;">
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
RNA-Seq
</td>
<td style="text-align:left;width: 30em; ">
sequencing of RNA molecules found in a population of cells or in a tissue
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
ChIP-Seq
</td>
<td style="text-align:left;width: 30em; ">
sequencing of DNA regions that are bound to particular DNA-binding proteins (selected by immunoprecipitation)
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
RIP-Seq
</td>
<td style="text-align:left;width: 30em; ">
sequencing of RNA molecules, or regions of them, bound to a particular RNA-binding protein
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
DNA-Seq
</td>
<td style="text-align:left;width: 30em; ">
sequencing of genomic DNA
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
HiC
</td>
<td style="text-align:left;width: 30em; ">
high-throughput chromatin conformation capture; a technique that aims to map the 3D spatial arrangement of DNA
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
cDNA
</td>
<td style="text-align:left;width: 30em; ">
complementary DNA made from RNA templates and reverse transcriptase; used in RNA-Seq
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
genetic screens
</td>
<td style="text-align:left;width: 30em; ">
a technique looking at the proliferation or survival of cells upon gene knockdown, knockout, or modification
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
read
</td>
<td style="text-align:left;width: 30em; ">
the sequence obtained from a fragment
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
sequencing library
</td>
<td style="text-align:left;width: 30em; ">
the collection of DNA molecules used as input for the sequencing machine
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
fragments
</td>
<td style="text-align:left;width: 30em; ">
molecules being sequenced during a sequencing analysis
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
count table
</td>
<td style="text-align:left;width: 30em; ">
a matrix with the tallies of the number of occurrences of subpopulations from a larger population/sample
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
dynamic range
</td>
<td style="text-align:left;width: 30em; ">
a ratio between the maximum and minimum values
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
heteroskedasticity
</td>
<td style="text-align:left;width: 30em; ">
a phenomenon where the variance and distribution shape of the data in different parts of the dynamic range are very different
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
normalization
</td>
<td style="text-align:left;width: 30em; ">
a technique that adjusts for the nature and magnitude of systematic sampling biases
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
rare events
</td>
<td style="text-align:left;width: 30em; ">
occurrences in the tail(s) of a distribution; observations that are extraordinarily high or low
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
dispersion
</td>
<td style="text-align:left;width: 30em; ">
a measure of the spread of the data; a common measure is the standard deviation or variance
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
gamma-Poisson
</td>
<td style="text-align:left;width: 30em; ">
negative binomial distribution with 2 parameters; 𝛼 and 𝛽
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
systematic biases
</td>
<td style="text-align:left;width: 30em; ">
systematic distortions that affect the data generation and need to be accounted for in the analysis; one example would be variations in the total number of reads for each sample in a sequencing experiment
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
metadata
</td>
<td style="text-align:left;width: 30em; ">
a set of data that describes or gives information about other data
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
multifactorial design
</td>
<td style="text-align:left;width: 30em; ">
an experimental design with more than one independent variable
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
balanced
</td>
<td style="text-align:left;width: 30em; ">
in the context of study design, these are where there is an equal number of observations of all combinations of factors being tested
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
differential expression analysis
</td>
<td style="text-align:left;width: 30em; ">
a type of analysis that uses the normalized read count data to investigate quantitative changes in expression levels between different experimental groups
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
intercept
</td>
<td style="text-align:left;width: 30em; ">
a coefficient representing the base level of the measurement in the negative control
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
design factors
</td>
<td style="text-align:left;width: 30em; ">
binary indicator variables
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
interaction effect
</td>
<td style="text-align:left;width: 30em; ">
a parameter in a model that accounts for the effects of two experimental factors that combine in a more complicated fashion than a simple summation
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
design matrix
</td>
<td style="text-align:left;width: 30em; ">
a matrix encoding the design of an experiment where the columns correspond to experimental factors and the rows correspond to different experimental conditions
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
residuals
</td>
<td style="text-align:left;width: 30em; ">
a term in a model that reflects the experimental fluctuations (i.e. random noise)
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
least sum-of-squares fitting
</td>
<td style="text-align:left;width: 30em; ">
a type of model fitting that minimizes the sum of the squared residuals
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
linear model
</td>
<td style="text-align:left;width: 30em; ">
a model that is a linear function of parameters, i.e. takes the form: y_j = sum_k (x_jk * beta_k + e_j)
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
analysis of variance (ANOVA)
</td>
<td style="text-align:left;width: 30em; ">
an analysis that decomposes patterns in the data into systematic variability and noise
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
noise
</td>
<td style="text-align:left;width: 30em; ">
variability unaccounted for by model parameters
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
systematic variability
</td>
<td style="text-align:left;width: 30em; ">
variability accounted for by model parameters
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
breakdown point
</td>
<td style="text-align:left;width: 30em; ">
a measure of the robustness of an estimator; larger values indicate more robust estimators
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
robust
</td>
<td style="text-align:left;width: 30em; ">
a “sturdy” estimator that is not heavily influenced by outliers
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
least absolute deviations
</td>
<td style="text-align:left;width: 30em; ">
minimization of the sum of the absolute values of the residuals
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
least quantile of squares
</td>
<td style="text-align:left;width: 30em; ">
a type of regression where the difference between the model quantile and empirical quantile is minimized
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
least trimmed sum of squares
</td>
<td style="text-align:left;width: 30em; ">
a type of regression that minimized the sum of squared residuals, where the sum is over only a fraction of the smallest residuals
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
logistic regression
</td>
<td style="text-align:left;width: 30em; ">
a type of generalized linear regression for binary data where the outcome is transformed by the logistic function and bounded between 0 and 1
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
maximum likelihood
</td>
<td style="text-align:left;width: 30em; ">
a method for parameter estimation that finds the parameter value that maximizes the probablity of the observed data under the model
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
likelihood
</td>
<td style="text-align:left;width: 30em; ">
a function of a model parameter which is equal to the probability of the observed data under the model
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
maximum-likelihood estimates
</td>
<td style="text-align:left;width: 30em; ">
model parameters that are estimated by maximizing the probability of the observed data under the model
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
nuisance factor / blocking factor
</td>
<td style="text-align:left;width: 30em; ">
a factor that has some effect on the response but is of no interest to the experiment
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
batch effects
</td>
<td style="text-align:left;width: 30em; ">
hidden factors that affect the data but are not documented; e.g. running samples at the same time have a degree of similarity from being run in the same batch
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
pseudocounts
</td>
<td style="text-align:left;width: 30em; ">
transformations that take the form y = log2(n + n_0) where n is the count and n_0 is a chosen positive constant
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
variance stabilizing transformation
</td>
<td style="text-align:left;width: 30em; ">
a transformation that has finite values and finite slope, even for counts close to zero
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
regularized logarithm (rlog) transformation
</td>
<td style="text-align:left;width: 30em; ">
a technique that transforms the original count data to a log2-like scale by fitting a “trivial” model with a separate term for each sample and a prior distribution on the coefficients which is estimated from the data
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
Cook’s distance
</td>
<td style="text-align:left;width: 30em; ">
a measure of how much a single sample is influencing the coefficients in a model; large values indicate an outlier count
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
sampling without replacement
</td>
<td style="text-align:left;width: 30em; ">
a random sample in which no observation occurs more than one time in the sample
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
null hypothesis
</td>
<td style="text-align:left;width: 30em; ">
often, a hypothesis of “no association” that is used as a counterpart to a more interesting alternative hypothesis in hypothesis testing.
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
variability
</td>
<td style="text-align:left;width: 30em; ">
in statistics, the amount by which a set of observations deviate from their mean
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
outlier
</td>
<td style="text-align:left;width: 30em; ">
a data point that does not follow the pattern of the rest of the data; often this data point will have a large residual
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
M-estimation
</td>
<td style="text-align:left;width: 30em; ">
a type of regression analysis that is more robust than OLS to outliers or data that does not follow a normal distribution; it minimizes the sum of the penalization function applied to the residuals
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
conservative
</td>
<td style="text-align:left;width: 30em; ">
an approach that prioritizes reducing false positives
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
splicing
</td>
<td style="text-align:left;width: 30em; ">
a process in eukaryotic organisms where mRNA is cut down from the full-length gene to just the exons before being translated
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
exons
</td>
<td style="text-align:left;width: 30em; ">
segments of a gene that actually get used during translation or encode for a protein
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
isoforms
</td>
<td style="text-align:left;width: 30em; ">
different forms of the same gene that result from splicing events that combine different exons in an mRNA script
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
upregulated
</td>
<td style="text-align:left;width: 30em; ">
a term used to describe the increased expression of a gene
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
gene knockdown
</td>
<td style="text-align:left;width: 30em; ">
a way of inactivating a gene by targeting its mRNA transcript for inactivation or degradation
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
gene knockout
</td>
<td style="text-align:left;width: 30em; ">
deletion of a gene from the genome
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
transcriptome
</td>
<td style="text-align:left;width: 30em; ">
the total of all of the mRNA expressed from genes in an organism
</td>
</tr>
<tr>
<td style="text-align:right;font-weight: bold;border-right:1px solid;">
polymorphism
</td>
<td style="text-align:left;width: 30em; ">
genetic variation within a population
</td>
</tr>
</tbody>
</table>
<div id="sources-consulted-or-cited" class="section level3">
<h3>Sources consulted or cited</h3>
<p><em>Some of the definitions above are based in part or whole on listed definitions in the following sources.</em></p>
<ul>
<li>Holmes and Huber, 2019. <em>Modern Statistics for Modern Biology.</em> Cambridge University Press,
Cambridge, United Kingdom.</li>
<li>Lexico: <a href="https://www.lexico.com" class="uri">https://www.lexico.com</a></li>
<li>Statistics How To: <a href="https://www.statisticshowto.com" class="uri">https://www.statisticshowto.com</a></li>
<li>Lavrakas, 2008. Sampling without replacement. <em>Encyclopedia of Survey Research Methods.</em> <a href="https://dx.doi.org/10.4135/9781412963947.n516" class="uri">https://dx.doi.org/10.4135/9781412963947.n516</a></li>
</ul>
</div>
<div id="practice" class="section level3">
<h3>Practice</h3>
<iframe src="https://quizlet.com/501110916/flashcards/embed?i=2shf5e&amp;x=1jj1" height="500" width="100%" style="border:0">
</iframe>
</div>

    </div>

    





<div class="article-tags">
  
  <a class="badge badge-light" href="/tags/vocabulary/">vocabulary</a>
  
  <a class="badge badge-light" href="/tags/chapter-8/">chapter 8</a>
  
</div>



<div class="share-box" aria-hidden="true">
  <ul class="share">
    
      
      
      
        
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=/post/chapter-8-vocabulary-list/&amp;text=Chapter%208%20Vocabulary%20List" target="_blank" rel="noopener" class="share-btn-twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=/post/chapter-8-vocabulary-list/&amp;t=Chapter%208%20Vocabulary%20List" target="_blank" rel="noopener" class="share-btn-facebook">
          <i class="fab fa-facebook"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="mailto:?subject=Chapter%208%20Vocabulary%20List&amp;body=/post/chapter-8-vocabulary-list/" target="_blank" rel="noopener" class="share-btn-email">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=/post/chapter-8-vocabulary-list/&amp;title=Chapter%208%20Vocabulary%20List" target="_blank" rel="noopener" class="share-btn-linkedin">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://web.whatsapp.com/send?text=Chapter%208%20Vocabulary%20List%20/post/chapter-8-vocabulary-list/" target="_blank" rel="noopener" class="share-btn-whatsapp">
          <i class="fab fa-whatsapp"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://service.weibo.com/share/share.php?url=/post/chapter-8-vocabulary-list/&amp;title=Chapter%208%20Vocabulary%20List" target="_blank" rel="noopener" class="share-btn-weibo">
          <i class="fab fa-weibo"></i>
        </a>
      </li>
    
  </ul>
</div>












  
  
    
  
  






  
  
  
  
  
  <div class="media author-card content-widget-hr">
    
      
      <img class="avatar mr-3 avatar-circle" src="/authors/mikaela-elder/avatar_hu785d7f79b40e2e21c1510df369f80c58_1881166_270x270_fill_q90_lanczos_center.jpg" alt="Avatar">
    

    <div class="media-body">
      <h5 class="card-title"><a href="/authors/mikaela-elder/">Mikaela Elder</a></h5>
      <h6 class="card-subtitle">Undergraduate Student in Biochemistry with Statistics Minor</h6>
      <p class="card-text">I'm an undergraduate student interested in learning how to mathematically model biological systems.</p>
      <ul class="network-icon" aria-hidden="true">
  
    
    
    
      
    
    
    
    
    
    <li>
      <a href="mailto:melder@rams.colostate.edu" >
        <i class="fas fa-envelope"></i>
      </a>
    </li>
  
</ul>

    </div>
  </div>









  
  
  <div class="article-widget content-widget-hr">
    <h3>Related</h3>
    <ul>
      
      <li><a href="/post/vocabulary-for-chapter-7/">Vocabulary for Chapter 7</a></li>
      
      <li><a href="/post/vocabulary-for-chapter-6/">Vocabulary for Chapter 6</a></li>
      
      <li><a href="/post/vocabulary-for-chapter-5/">Vocabulary for Chapter 5</a></li>
      
      <li><a href="/post/vocabulary-for-chapter-4/">Vocabulary for Chapter 4</a></li>
      
      <li><a href="/post/vocabulary-for-chapter-2-8-2-12/">Vocabulary for Chapter 2, Part 2</a></li>
      
    </ul>
  </div>
  



  </div>
</article>

      

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js" integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js" integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.js" integrity="sha256-yt2kYMy0w8AbtF89WXb2P1rfjcP/HTHLT7097U8Y5b8=" crossorigin="anonymous"></script>

      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mermaid/8.4.3/mermaid.min.js" integrity="" crossorigin="anonymous" title="mermaid"></script>
      

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.10/highlight.min.js" integrity="sha256-1zu+3BnLYV9LdiY85uXMzii3bdrkelyp37e0ZyTAQh0=" crossorigin="anonymous"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.10/languages/r.min.js"></script>
        
      

    

    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.5.1/leaflet.js" integrity="sha256-EErZamuLefUnbMBQbsEqu1USa+btR2oIlCpBJbyD4/g=" crossorigin="anonymous"></script>
    

    
    
    <script>const code_highlighting = true;</script>
    

    
    
    
    
    
    
    <script>
      const search_config = {"indexURI":"/index.json","minLength":1,"threshold":0.3};
      const i18n = {"no_results":"No results found","placeholder":"Search...","results":"results found"};
      const content_type = {
        'post': "Posts",
        'project': "Projects",
        'publication' : "Publications",
        'talk' : "Talks"
        };
    </script>
    

    
    

    
    
    <script id="search-hit-fuse-template" type="text/x-template">
      <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
      </div>
    </script>
    

    
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js" integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>
    

    
    

    
    

    
    
    
    
    
    
    
    
    
      
    
    
    
    
    <script src="/js/academic.min.a0d331bcd05dbe8b31e244f796710f08.js"></script>

    






  
  
  <div class="container">
    <footer class="site-footer">
  
  <p class="powered-by">
    
      <a href="/privacy/">Privacy Policy</a>
    
    
       &middot; 
      <a href="/terms/">Terms</a>
    
  </p>
  

  <p class="powered-by">
    

    Powered by the
    <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
    <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

    
    <span class="float-right" aria-hidden="true">
      <a href="#" class="back-to-top">
        <span class="button_icon">
          <i class="fas fa-chevron-up fa-2x"></i>
        </span>
      </a>
    </span>
    
  </p>
</footer>

  </div>
  

  
<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <pre><code class="tex hljs"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

</body>
</html>
